modules/services/window-managers/i3-sway/lib/options.nix

```diff
@@ -390,8 +390,11 @@ in {
         };
 
         followMouse = mkOption {
-          type = types.bool;
-          default = true;
+          type = if moduleName == "sway" then
+            types.enum [ "yes" "no" "always" ]
```

Comment:
Best have `either bool (enum [ "yes" "no" "always" ])` to remain backwards compatible.

Reply:
Hm. What does the other side of that look like then, when rendering out the config?

Reply:
You can probably add something like

```
apply = p:
  if moduleName == "sway" && isBool p then
    (if p then "yes" else "no")
  else
    p;
```

and leave the rest of the code as-is. If the user assigns a boolean to the option it will be converted to "yes" or "no" in the `config` attribute set.

Reply:
I mean to use the [`apply` attribute](https://github.com/NixOS/nixpkgs/blob/72066757c8bd31cc592914cb944f8d7da569d741/lib/options.nix#L43-L44) to the `mkOption` function.

I'm quite confused how it could work when you tested using a literal `true` or `false` in your sway configuration with this change applied. I've been looking through the library code for a reason since it would seem like a bug to me.

Reply:
I understand now, and understand the missed test case. I'll update and check more exhaustively before pinging for another review. Thanks for the attention.

Reply:
Cool. To improve confidence in the change you could add a test case with the option set to a Boolean value and verify that it gets mapped to the enum value in the generated configuration. There is already similar test `i3-keybindings.nix`, which I guess you could copy and adjust as necessary.